.github/workflows/test-lang-rust-ci.yml

```diff
@@ -59,19 +60,27 @@ jobs:
         uses: actions/checkout@v4
 
       - name: Cache Cargo
+        id: cache-cargo
         uses: actions/cache@v3
         with:
           # these represent dependencies downloaded by cargo
           # and thus do not depend on the OS, arch nor rust version.
           path: ~/.cargo
-          key: cargo-cache1-
+          key: ${{ runner.os }}-cargo-cache1-${{ matrix.target }}-${{ matrix.rust }}-${{ hashFiles('**/Cargo.lock') }}
+
```

Comment:
I didn't use `${{ matrix.target }}` and `${{ matrix.rust }}` intentionally because many caches are generated and take up the cache capacity.
See https://github.com/apache/avro/actions/caches
Rust related cache takes up approximately 9GB while the capacity limit is 10GB.
Do you think it's cost effective for the degree of the improvement?

Reply:
If you don't mind, I'll update that PR with a compromised solution.